patches/tModLoader/Terraria/ModLoader/Config/ConfigManager.cs

```diff
@@ -72,6 +81,82 @@ internal static void Add(ModConfig config)
 		configList2.Add(GeneratePopulatedClone(config));
 	}
 
+	internal static void FinishSetup()
+	{
+		typesWithLocalizationRegistered.Clear();
```

Comment:
Best to do this in `Unload`, or register a callback to `TypeCaching.OnClear` (which is just called from `Unload`)